src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -7189,6 +7192,38 @@ struct at_sylvanas_windrunner_raze : AreaTriggerAI
     InstanceScript* _instance;
 };
 
+// Serverside - Sylvanas Windrunner's Position Z Check
+struct at_sylvanas_windrunner_z_check : AreaTriggerAI
+{
+    at_sylvanas_windrunner_z_check(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger),
+        _instance(at->GetInstanceScript()) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!_instance || !unit->IsAlive())
+            return;
+
+        if (Player* player = unit->ToPlayer())
+        {
+            if (player->IsGameMaster())
+                return;
+
+            if ((player->GetAreaId() == AREA_PINNACLE_OF_DOMINANCE && player->GetPositionZ() < 4100.5336f) ||
```

Comment:
Why do you check here again for PosZ when the area in db already ensures player is at a specific PosZ?

Reply:
True! I'll rely on DB now.